### ISSUE TITLE
abort() in python on windows debug nightly

### ISSUE THREAD

**Issue by nuclearsandwich** [ros2/build_farmer#188]
I spent a chunk of today investigating https://ci.ros2.org/view/nightly/job/nightly_win_deb/1243/

I started by reproducing the failure in https://ci.ros2.org/job/ci_windows/6824/ and then investigated the CI workspace. When running the ros2action tests manually I could repro and got a popup that python_d had abort()ed.

I eventually got as far as triggering the abort() just by running `rclpy.init()` from the python debug interpreter. I tried reverting a few recent rclpy packages, eventually reverting back to https://github.com/ros2/rclpy/commit/d340737bc3b1fd30435bf7d0fbcd41ea5018043e but the issue persisted. I needed to change to a more local host for further debugging but hit different issues and got bogged down. I've assigned myself to make sure that this gets attention but I'll be unavailable until next week so if someone else wants to take it feel free to kick me off.

**Comment by mjcarroll** [comment]
I believe that this is related to a larger set of test failures on Debug Windows builds. 

Over the last week, I discovered that the current `log4cxx` choco package had incorrectly-built `.lib/.dll` pairs for debug, which were pointing to the release `dll`, regardless of the build type.

I corrected the package and included it on the current release in the choco-packages repo: https://github.com/ros2/choco-packages/releases/tag/2019-02-15-1, as well as deployed it on the ROS2 CI machines.

After doing that, I surfaced some CMake issues, which need to be addressed via this PR: https://github.com/ros2/rcl_logging/pull/13
